src/core/src/validation_util.cpp

```diff
@@ -1069,37 +1071,103 @@ int64_t ov::util::clip(const int64_t& value, const int64_t& min, const int64_t&
-        return nullptr;
+    std::map<Output<Node>, std::shared_ptr<Node>> node_map;
+    std::stack<std::shared_ptr<Node>> stack;
+    stack.push(subgraph_sink.get_node_shared_ptr());
 
-    if (subgraph_sink.get_tensor().has_and_set_bound()) {
-        const auto& lower = subgraph_sink.get_tensor().get_lower_value();
-        return std::make_shared<ov::op::v0::Constant>(lower);
-    }
+    while (!stack.empty()) {
+        auto node = stack.top();
+        size_t num_inputs = node->get_input_size();
+        if (num_inputs == 0)
+            return nullptr;
 
-    if (ov::is_type<op::util::ShapeOfBase>(node) && node->get_input_partial_shape(0).is_dynamic()) {
-        return nullptr;
+        if (ov::pass::constant_folding_is_disabled(node))
+            return nullptr;
+
+        bool node_has_bounds_set = true;
+        for (size_t i = 0; i < node->get_output_size(); i++) {
+            const auto& tensor = node->get_output_tensor(i);
+            bool tensor_has_bounds_set = tensor.has_and_set_bound();
+            node_has_bounds_set = node_has_bounds_set && tensor_has_bounds_set;
+            if (tensor_has_bounds_set) {
+                const auto& lower = node->get_output_tensor(i).get_lower_value();
+                auto constant = std::make_shared<ov::op::v0::Constant>(lower);
+                node_map[node->output(i)] = constant;
+            }
+        }
+
+        if (node_has_bounds_set) {
+            stack.pop();
+            continue;
+        }
+
+        if (ov::is_type<op::util::ShapeOfBase>(node) && node->get_input_partial_shape(0).is_dynamic()) {
+            return nullptr;
+        }
+
+        auto original_node = node;
+
+        node = ov::util::convert_to_supported_precision(node, get_inputs_from_map(node, node_map));
+
+        OutputVector outputs(node->get_output_size());
+        if (node->constant_fold(outputs, node->input_values())) {
+            stack.pop();
+            for (size_t i = 0; i < outputs.size(); i++) {
+                node_map[original_node->output(i)] = outputs[i].get_node_shared_ptr();
+            }
+        } else {
+            bool inputs_pushed_on_stack = false;
+            for (size_t i = node->get_input_size(); i > 0; i--) {
+                auto input = original_node->input_value(i - 1);
+                if (node_map.count(input) == 0 && !ov::op::util::is_constant(input.get_node())) {
+                    stack.push(input.get_node_shared_ptr());
+                    inputs_pushed_on_stack = true;
+                }
+            }
+            // if none of the inputs was pushed to stack, it means the node that was not constantfolded
+            // is processed the second time. If that case - the node is not constfoldable.
+            // A good example would be a node that all of its inputs are constants and yet it cannot be constantfolded
+            // for some reason (like lack of evaluate, it's a op::util::FrameworkNode, etc.).
+            if (!inputs_pushed_on_stack) {
```

Comment:
compare current stack size with value before loop and remove bool variable set in the loop above.